app/controllers/votes_controller.rb

```diff
@@ -0,0 +1,22 @@
+class VotesController < ApplicationController
+  before_action :require_authentication
+
+  before_action :find_or_initialize_vote, only: %i[create destroy]
+
+  def create
+    redirect_to course_path(@vote.course_slug)
+  end
+
+  def destroy
+    @vote.destroy
+    redirect_to course_path(@vote.course_slug)
+  end
+
+  private
+
+  def find_or_initialize_vote
+    @vote = current_user.votes.find_or_initialize_by(course_slug: params[:course_id])
+    @vote.user_id = current_user.id
```

Comment:
Don't need this (as `user_id` will automatically be set in the line above).

Reply:
Done.